pkg/cmd/pulumi/plugin/plugin_install.go

```diff
@@ -125,6 +125,14 @@ func (cmd *pluginInstallCmd) Run(ctx context.Context, args []string) error {
 		cmd.registry = cmdCmd.NewDefaultRegistry(ctx, pkgWorkspace.Instance, nil, cmd.diag, cmd.env)
 	}
 
+	// If --server flag is not set, check for PULUMI_PLUGIN_SERVER environment variable
+	if cmd.serverURL == "" {
+		if serverFromEnv := env.PluginServer.Value(); serverFromEnv != "" {
+			cmd.serverURL = serverFromEnv
+			logging.V(1).Infof("Using plugin server from PULUMI_PLUGIN_SERVER: %s", serverFromEnv)
+		}
+	}
```

Comment:
I believe that this can already be set via `PULUMI_OPTION_SERVER`.

https://github.com/pulumi/pulumi/blob/ac51dada3968a3dc7d3598bab165da228e29f704/pkg/cmd/pulumi/pulumi.go#L940-L945

We don't need an additional env var here.